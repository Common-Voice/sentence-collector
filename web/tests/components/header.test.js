import React from 'react';
import { shallow } from 'enzyme';
import * as redux from 'react-redux';

import Header from '../../src/components/header';

beforeEach(() => {
  jest.clearAllMocks();
  jest.spyOn(redux, 'useSelector');
});

test('should render logged out state sidebar', () => {
  redux.useSelector.mockImplementation(() => ({ authed: false, migrationDone: false }));
  let component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});

test('should render logged in state sidebar', () => {
  redux.useSelector.mockImplementation(() => ({ authed: true, migrationDone: false }));
  let component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});
