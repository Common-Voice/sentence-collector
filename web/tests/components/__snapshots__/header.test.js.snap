// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render logged in state sidebar 1`] = `
<header>
  <Link
    href=""
    to="/"
  >
    <img
      src="cv-logo-one-color-white.svg"
    />
  </Link>
  <nav>
    <NavLink
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      exact={true}
      to="/how-to"
    >
      How-to
    </NavLink>
    <NavLink
      exact={true}
      key="add"
      to="/add"
    >
      Add
    </NavLink>
    <NavLink
      key="review"
      to="/review"
    >
      Review
    </NavLink>
    <NavLink
      key="rejected"
      to="/rejected"
    >
      Rejected Sentences
    </NavLink>
    <NavLink
      key="stats"
      to="/stats"
    >
      Statistics
    </NavLink>
    <NavLink
      exact={true}
      key="profile"
      to="/profile"
    >
      Profile
    </NavLink>
  </nav>
  <ProfileWidget />
  <section
    id="external-links"
  >
    <a
      href="https://discourse.mozilla.org/tags/c/voice/sentence-collection"
      rel="noopener noreferrer"
      target="_blank"
    >
      Discourse
    </a>
    <a
      href="https://github.com/Common-Voice/sentence-collector/issues"
      rel="noopener noreferrer"
      target="_blank"
    >
      Report Bugs (GitHub)
    </a>
    <a
      href="https://discourse.mozilla.org/t/sentence-collector-copyright-issues/52767"
      rel="noopener noreferrer"
      target="_blank"
    >
      Report copyright issues
    </a>
    <a
      href="https://commonvoice.mozilla.org/privacy"
      rel="noopener noreferrer"
      target="_blank"
    >
      Privacy
    </a>
    <a
      href="https://commonvoice.mozilla.org/terms"
      rel="noopener noreferrer"
      target="_blank"
    >
      Terms
    </a>
    <a
      href="https://www.mozilla.org/en-US/privacy/websites/#cookies"
      rel="noopener noreferrer"
      target="_blank"
    >
      Cookies
    </a>
  </section>
</header>
`;

exports[`should render logged out state sidebar 1`] = `
<header>
  <Link
    href=""
    to="/"
  >
    <img
      src="cv-logo-one-color-white.svg"
    />
  </Link>
  <nav>
    <NavLink
      exact={true}
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      exact={true}
      to="/how-to"
    >
      How-to
    </NavLink>
    <NavLink
      exact={true}
      key="add"
      to="/add"
    >
      Add
    </NavLink>
    <NavLink
      key="review"
      to="/review"
    >
      Review
    </NavLink>
    <NavLink
      key="rejected"
      to="/rejected"
    >
      Rejected Sentences
    </NavLink>
    <NavLink
      key="stats"
      to="/stats"
    >
      Statistics
    </NavLink>
    <a
      href="/sentence-collector/login"
    >
      Login
    </a>
  </nav>
  <ProfileWidget />
  <section
    id="external-links"
  >
    <a
      href="https://discourse.mozilla.org/tags/c/voice/sentence-collection"
      rel="noopener noreferrer"
      target="_blank"
    >
      Discourse
    </a>
    <a
      href="https://github.com/Common-Voice/sentence-collector/issues"
      rel="noopener noreferrer"
      target="_blank"
    >
      Report Bugs (GitHub)
    </a>
    <a
      href="https://discourse.mozilla.org/t/sentence-collector-copyright-issues/52767"
      rel="noopener noreferrer"
      target="_blank"
    >
      Report copyright issues
    </a>
    <a
      href="https://commonvoice.mozilla.org/privacy"
      rel="noopener noreferrer"
      target="_blank"
    >
      Privacy
    </a>
    <a
      href="https://commonvoice.mozilla.org/terms"
      rel="noopener noreferrer"
      target="_blank"
    >
      Terms
    </a>
    <a
      href="https://www.mozilla.org/en-US/privacy/websites/#cookies"
      rel="noopener noreferrer"
      target="_blank"
    >
      Cookies
    </a>
  </section>
</header>
`;
